### ISSUE TITLE
Performance/stress tests for Akka.IO

### ISSUE THREAD

**Issue by Horusiath** [akkadotnet/akka.net#2414]
From what I've checked, *akka-io* branch (see PR: #2405) already passes more of the existing specs (that includes Akka.Streams integration) than the current version of components in Akka.IO namespace.

However this doesn't mean that it's ready. Back in the days when we initially released Akka.IO, we pretty fast hit multiple bugs related to high CPU usage or memory issues. This was something we haven't got ready tests against. 

For this we need to prepare some stress tests, that will check usual things like how many concurrent connections are we able to operate on or how much throughput are we able to handle before we crush the application. This also needs to include CPU and memory assertions. The starting point here could be something @Aaronontheweb [already did for Helios](https://github.com/helios-io/helios/blob/dev/benchmark/Helios.HorizontalScaling.Performance/Channels/TcpServerSocketChannelHorizontalScaleSpec.cs).

**Comment by alexvaluyskiy** [comment]
@Horusiath what about this one?

**Comment by alexvaluyskiy** [comment]
@Horusiath have we finished it?

**Comment by Horusiath** [comment]
No, I have some tests of mine, but they are running as standalone applications and are not part of test suite.

**Comment by Aaronontheweb** [comment]
Done - scaled Akka.IO to 1m concurrent connections this weekend in a single process and we have solid benchmarks for it now.
